mpp/include/mpp_write_2Ddecomp.h

```diff
@@ -160,9 +160,27 @@
               write( unit,* )field%id, subdomain, time_level, time, data
           else                      !MPP_IEEE32 or MPP_NATIVE
               if( mpp_file(unit)%access.EQ.MPP_SEQUENTIAL )then
+#ifdef __sgi
+                  if( mpp_file(unit)%format.EQ.MPP_IEEE32 )then
+                      data_r4 = data !IEEE conversion layer on SGI until assign -N ieee_32 is supported
+                      write(unit)field%id, subdomain, time_level, time, data_r4
+                  else
+                      write(unit)field%id, subdomain, time_level, time, data
+                  end if
+#else
                   write(unit)field%id, subdomain, time_level, time, data
+#endif
               else                  !MPP_DIRECT
+#ifdef __sgi
+                  if( mpp_file(unit)%format.EQ.MPP_IEEE32 )then
+                      data_r4 = data !IEEE conversion layer on SGI until assign -N ieee_32 is supported
+                      write( unit, rec=mpp_file(unit)%record )field%id, subdomain, time_level, time, data_r4
+                  else
+                      write( unit, rec=mpp_file(unit)%record )field%id, subdomain, time_level, time, data
+                  end if
+#else
```

Comment:
These lines were removed as part of the macro cleanup and need to be removed here.